internal/dot/graph.go

```diff
@@ -252,6 +270,50 @@ func (dg *Graph) addToGroup(r *Result, id CtorID) {
 	group.Results = append(group.Results, r)
 }
 
+// Prune removes elements from the graph that do not have failed results.
+// Removing elements that do not have failing results makes the graph easier to debug, since non-failing
+// nodes and edges can clutter the graph and don't help the user debug.
+func (dg *Graph) Prune() {
+	failedResults := make(map[*Result]bool)
```

Comment:
If possible, I would avoid using `*Result` as map keys. Using pointers as map
keys can lead to confusing/unexpected behavior.

For example, https://play.golang.org/p/x69eTtwuMLz. (This is probably the
behavior we want here but it's easier to mess it up by using it as a map key.)

You can define a groupKey-like struct instead.

Reply:
Resolved.

Reply:
Resolved